Formula/tendermint.rb

```diff
@@ -0,0 +1,20 @@
+class Tendermint < Formula
+  desc "⟁ Tendermint Core (BFT Consensus) in Go"
+  homepage "https://tendermint.com/"
+  url "https://github.com/tendermint/tendermint/archive/v0.34.0.tar.gz"
+  sha256 "3a28fac4c5e610fc32763db1b717ec0a1e12d39262e321d6da223c3b0acfea7f"
+  license "Apache-2.0"
+  depends_on "go" => :build
+  head "https://github.com/tendermint/tendermint.git"
+
+  def install
+    system "make", "tools"
+    system "make", "build", "LDFLAGS=-X github.com/tendermint/tendermint/version.TMCoreSemVer=#{version}"
+    bin.install "build/tendermint"
+  end
+
+  test do
+    tendermint_version_without_hash = shell_output("#{bin}/tendermint version").partition("-").first
```

Comment:
I don't remember the rules here exactly, but printing version may not be sufficient. What do you think about actually running the binary `TMHOME=/tmp tendermint init && tendermint node --proxy_app=kvstore` ?

Reply:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

Reply:
yup that sounds like a good idea. What do you want me to assert exactly? that it doesn't exit with an error?

Reply:
let's assert that tendermint produces blocks